src/core/textrenderer/qgstextcharacterformat.cpp

```diff
@@ -54,6 +54,12 @@ QgsTextCharacterFormat::QgsTextCharacterFormat( const QTextCharFormat &format )
   , mOverline( format.hasProperty( QTextFormat::FontOverline ) ? ( format.fontOverline() ? BooleanValue::SetTrue : BooleanValue::SetFalse ) : BooleanValue::NotSet )
 {
   mVerticalAlign = convertTextCharFormatVAlign( format, mHasVerticalAlignSet );
+  if ( mFontFamily.isEmpty() && format.hasProperty( QTextFormat::FontFamilies ) )
```

Comment:
I feel the code would be more readable if you'd move the `, mFontFamily( format.hasProperty( QTextFormat::FontFamily ) ? format.fontFamily() : QString() )` line above into the main function code. Otherwise looks good, and thanks, I've stumbled on that issue while running Qt6 builds.

Reply:
Done!